packages/contracts-bedrock/test/setup/Setup.sol

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { L2CrossDomainMessenger } from "src/L2/L2CrossDomainMessenger.sol";
+import { L2StandardBridge } from "src/L2/L2StandardBridge.sol";
+import { L2ToL1MessagePasser } from "src/L2/L2ToL1MessagePasser.sol";
+import { L2ERC721Bridge } from "src/L2/L2ERC721Bridge.sol";
+import { BaseFeeVault } from "src/L2/BaseFeeVault.sol";
+import { SequencerFeeVault } from "src/L2/SequencerFeeVault.sol";
+import { L1FeeVault } from "src/L2/L1FeeVault.sol";
+import { GasPriceOracle } from "src/L2/GasPriceOracle.sol";
+import { L1Block } from "src/L2/L1Block.sol";
+import { LegacyMessagePasser } from "src/legacy/LegacyMessagePasser.sol";
+import { GovernanceToken } from "src/governance/GovernanceToken.sol";
+import { OptimismMintableERC20Factory } from "src/universal/OptimismMintableERC20Factory.sol";
+import { LegacyERC20ETH } from "src/legacy/LegacyERC20ETH.sol";
+import { StandardBridge } from "src/universal/StandardBridge.sol";
+import { FeeVault } from "src/universal/FeeVault.sol";
+import { OptimismPortal } from "src/L1/OptimismPortal.sol";
+import { L1CrossDomainMessenger } from "src/L1/L1CrossDomainMessenger.sol";
+import { DeployConfig } from "scripts/DeployConfig.s.sol";
+import { Deploy } from "scripts/Deploy.s.sol";
+import { L2OutputOracle } from "src/L1/L2OutputOracle.sol";
+import { ProtocolVersions } from "src/L1/ProtocolVersions.sol";
+import { SystemConfig } from "src/L1/SystemConfig.sol";
+import { L1StandardBridge } from "src/L1/L1StandardBridge.sol";
+import { AddressManager } from "src/legacy/AddressManager.sol";
+import { L1ERC721Bridge } from "src/L1/L1ERC721Bridge.sol";
+import { AddressAliasHelper } from "src/vendor/AddressAliasHelper.sol";
+
+/// @title Setup
+/// @dev This contact is responsible for setting up the contracts in state. It currently
+///      sets the L2 contracts directly at the predeploy addresses instead of setting them
+///      up behind proxies. In the future we will migrate to importing the genesis JSON
+///      file that is created to set up the L2 contracts instead of setting them up manually.
+contract Setup is Deploy {
+    OptimismPortal optimismPortal;
+    L2OutputOracle l2OutputOracle;
+    SystemConfig systemConfig;
+    L1StandardBridge l1StandardBridge;
+    L1CrossDomainMessenger l1CrossDomainMessenger;
+    AddressManager addressManager;
+    L1ERC721Bridge l1ERC721Bridge;
+    OptimismMintableERC20Factory l1OptimismMintableERC20Factory;
+    ProtocolVersions protocolVersions;
+
+    L2CrossDomainMessenger l2CrossDomainMessenger;
```

Comment:
Suggestion: 
```suggestion
    L2CrossDomainMessenger constant l2CrossDomainMessenger =
        L2CrossDomainMessenger(payable(Predeploys.L2_CROSS_DOMAIN_MESSENGER));
```

Which then enables deleting quite a few lines in the `L2()` method below.

Reply:
Or if you don't like how this looks, you can make them `immutable` and set these in the constructor.